### PR TITLE
#612 - Display Name

### PR DESCRIPTION
**Pull request by jmarrec**
Just a proof of concept for now. Just on Space tab.

* #612

image

I suppose we should have a toggle switch somewhere in the preferences...

### CODE REVIEW

**Comment by macumber** [comment]
That looks great, I like a toggle but maybe on the tab itself?

**Comment by jmarrec** [comment]
Theoretically speaking, this applies to space, surfaces, shading surfaces, thermal zones, etc etc.
A toggle on each of these subtabs would be a hassle no?

File: src/openstudio_lib/OSDocument.cpp [hunk 1]
```diff
@@ -441,6 +444,7 @@ void OSDocument::createTab(int verticalId) {
   m_verticalId = verticalId;
 
   bool isIP = m_mainWindow->displayIP();
+  bool displayAdditionalProps = m_mainWindow->displayAdditionalProps();
```
Comment (macumber) [inline src/openstudio_lib/OSDocument.cpp:447]: Does `toggleDisplayAdditionalPropsClicked` get emitted somewhere on startup if it is set?

File: src/openstudio_lib/FacilityShadingGridView.cpp [hunk 2]
```diff
@@ -498,6 +507,22 @@ void FacilityShadingGridController::addColumns(const QString& category, std::vec
                           boost::optional<std::function<void(model::ShadingSurface*)>>(
                             std::function<void(model::ShadingSurface*)>([](model::ShadingSurface* t_ss) { t_ss->remove(); })),
                           boost::optional<std::function<bool(model::ShadingSurface*)>>(), DataSource(allShadingSurfaces, true));
+      } else if (field == DISPLAYNAME) {
+        addLoadNameColumn(Heading(QString(DISPLAYNAME), false, false),                                        // heading
```
Comment (macumber) [inline src/openstudio_lib/FacilityShadingGridView.cpp:511]: Maybe just `addValueEditColumn` would work here?  Or `addNameLineEditColumn`?  To show the value without the icon?

File: src/openstudio_lib/OSDocument.cpp (hunk 1 above)
Comment (jmarrec) [reply src/openstudio_lib/OSDocument.cpp:447]: It's read from settings and passed around. I dont think it needs to be emitted

File: src/openstudio_lib/FacilityShadingGridView.cpp (hunk 2 above)
Comment (jmarrec) [reply src/openstudio_lib/FacilityShadingGridView.cpp:511]: Yeah probably. I'll look into it

**Pull request merged**